src/video-interface.js

```diff
@@ -0,0 +1,165 @@
+   * the viewport, AMP will play the video and later pauses it when 25%
+   * or more of the video exits the viewport. If an auto-playing video also has
+   * controls (ie. `hasControls()` returns `true`), AMP will install a tap
+   * handler on the video, and when an end-user taps the video, AMP will show
+   * the controls.
+   *
+   * @return {boolean}
+   */
+  hasAutoplay() {}
+
+  /**
+   * Plays the video and returns a promise that will be resolved when the video
+   * is playing or rejected if the video could not be played.
+   *
+   * @param {boolean} isAutoplay Whether the call to the `play` method is
+   * triggered by the autoplay functionality. Video players can use this hint
+   * to make decisions such as not playing pre-roll video ads.
+   * @return {!Promise}
+   */
+  play(unusedIsAutoplay) {}
+
+
+  /**
+   * Pauses the video and returns a promise that will be resolved when the video
+   * is paused or rejected if the video could not be paused.
+   *
+   * @return {!Promise}
+   */
+  pause() {}
+
+  /**
+   * Mutes the video and returns a promise that will be resolved when the video
+   * is muted or rejected if the video could not be muted.
+   *
+   * @return {!Promise}
+   */
+  mute() {}
+
+  /**
+   * Unmutes the video and returns a promise that will be resolved when the
+   * video is muted or rejected if the video could not be unmuted.
+   *
+   * @return {!Promise}
+   */
+  unmute() {}
+
+  /**
+   * Whether the developer has configured the component to show UI controls such
+   * as play, pause, etc... buttons.
+   * This is normally done by setting `controls` attribute on the component.
+   *
+   * AMP runtime makes certain assumptions based on the value returned from
+   * `hasControls()` such as whether to allow end-users to interact with an
+   * auto-playing video or not.
+   */
+  hasControls() {}
+
+  /**
+   * Makes the video UI controls visible and returns a promise that will be
+   * resolved when the controls are displayed.
+   *
+   * AMP will not call this method if `hasControls()` returns false.
+   *
+   * @return {!Promise}
+   */
+  showControls() {}
+
+  /**
+   * Hides the video UI controls and returns a promise that will resolved when
+   * the controls are hidden.
+   *
+   * AMP will not call this method if `hasControls()` returns false.
+   *
+   * @return {!Promise}
+   */
+  hideControls() {}
+}
+
+/**
+ * Events
+ *
+ * Components implementing the VideoInterface are expected to dispatch
+ * the following DOM events.
+ *
+ * @constant {!Object<string, string>}
+ */
+export const VideoEvents = {
+  /**
+   * Built event, fired when component's `buildCallback` is finished.
+   *
+   * @event amp:video:built
+   */
+  BUILT: 'amp:video:built',
+
+  /**
+   * Canplay event, fired when the video player can start playing the video.
+   * Normally fired from `layoutCallback`.
+   *
+   * @event amp:video:canplay
+   */
+  CAN_PLAY: 'amp:video:built',
```

Comment:
Can play is the only video-specific event here. But I believe we just should use the existing video events and just polyfill them for all other players: https://developer.mozilla.org/en-US/docs/Web/Guide/Events/Media_events